snmp/datadog_checks/snmp/snmp.py

```diff
@@ -380,6 +383,12 @@ def _check_with_config(self, config):
                 error = 'Failed to collect metrics for {} - {}'.format(self._get_instance_name(instance), e)
             self.warning(error)
         finally:
+            # At this point, `tags` might include some extra tags added in try clause
+
+            # Sending `snmp.devices_monitored` with value 1 will allow users to count devices
+            # by using `sum by {X}` queries in UI. X being a tag like `autodiscovery_subnet`, `snmp_profile`, etc
+            self.gauge('snmp.devices_monitored', 1, tags=tags)
```

Comment:
I mentioned it already, but I'm not sure it's worth adding a new metric compared to just count `sysUptimeInstance`.

Reply:
Discussed on slack: `count_not_null(sum:snmp.sysUpTimeInstance{*} by {snmp_device})` works for counting the number of total devices, but it seems there is no way to count the number of devices per subnet (or any other tag like snmp_profile).

Reply:
Pros/Cons of the two ways:

1/ Use snmp.sysUpTimeInstance that is already available.
```
count:snmp.sysUpTimeInstance{*} by {autodiscovery_subnet} 
```

Pros:
- No new metric.

Cons:
- We count devices using a proxy metric snmp.sysUpTimeInstance . Less understandable by user. And create some coupling (devices count coupled with sysUpTimeInstance metric).
- count is currently a hidden feature and can only be used by editing the query

2/ Use a new metric called snmp.devices_monitored
```
sum:snmp.devices_monitored{*} by {autodiscovery_subnet}
```

Pros:
- Easier to understand and user friendly.

Cons:
- Need a new metric (payload/cost/networktraffic).